src/main/java/hillelee/homework/doctor_api/DoctorController.java

```diff
@@ -0,0 +1,71 @@
+package hillelee.homework.doctor_api;
+
+import lombok.AllArgsConstructor;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.net.URI;
+import java.util.Collection;
+import java.util.Optional;
+
+/**
+ * Created by 4oc3p on 30.11.2017. hillelee
+ */
+@RestController
+@AllArgsConstructor
+public class DoctorController {
+    private DoctorRepository doctorRepository;
+    private DoctorService doctorService;
+
+
+    @GetMapping(value = "/doctors")
+    public ResponseEntity<Collection<Doctor>> getAllDoctors(@RequestParam Optional<String> specialization,
+                                                            @RequestParam Optional<Character> name) {
+        return ResponseEntity.ok(
+                specialization.isPresent()
+                        ? doctorRepository.getDoctorsBySpecialization(specialization.get())
+                        : name.isPresent()
+                        ? doctorRepository.getDoctorsByFirstLetterOfName(name.get())
+                        : doctorRepository.getAllDoctors()
```

Comment:
First of all it is completely unreadable. Try to avoid ternary.
Next: what if request contains both specialization and name?